include/os/freebsd/spl/sys/list.h

```diff
@@ -58,6 +58,7 @@ int list_is_empty(list_t *);
 void list_link_init(list_node_t *);
 void list_link_replace(list_node_t *, list_node_t *);
 
+int list_link_not_active(list_node_t *);
```

Comment:
The FreeBSD implementation appears to be missing.

Reply:
I went ahead and added the the FreeBSD implementation of `list_link_not_active()`. I also added a couple more ASSERT checks to make sure we are not adding a gang ABD as a child to another gang, and that once a child is added to the gang the link shows up as active.